### ISSUE TITLE
[Issue]: TOT Build failure with Public release - ROCm 6.2.2

### ISSUE THREAD

**Issue by kiritigowda** [ROCm/rocPyDecode#100]
### Problem Description

```
sudo pip3 install .
Processing /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode
  Installing build dependencies ... done
  Getting requirements to build wheel ... error
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [130 lines of output]
      -- The CXX compiler identification is GNU 11.4.0
      -- Detecting CXX compiler ABI info
      -- Detecting CXX compiler ABI info - done
      -- Check for working CXX compiler: /usr/bin/c++ - skipped
      -- Detecting CXX compile features
      -- Detecting CXX compile features - done
      --      -D ROCPYDECODE_AMDGPU_TARGET= [Build for specific AMD GPUs]
      -- rocPyDecode -- AMDGPU_TARGETS: gfx908;gfx90a;gfx940;gfx941;gfx942;gfx1030;gfx1031;gfx1032;gfx1100;gfx1101;gfx1102
      -- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2")
      -- Checking for module 'libavcodec'
      --   Found libavcodec, version 58.134.100
      -- Checking for module 'libavformat'
      --   Found libavformat, version 58.76.100
      -- Checking for module 'libavutil'
      --   Found libavutil, version 56.70.100
      -- Using FFMPEG --
          Libraries:/usr/lib/x86_64-linux-gnu/libavcodec.so;/usr/lib/x86_64-linux-gnu/libavformat.so;/usr/lib/x86_64-linux-gnu/libavutil.so
          Includes:/usr/include/x86_64-linux-gnu
      -- NOTE: rocPyDecode requires Python3, Found --  /usr/bin/python3.10 
      -- Performing Test HAS_FLTO
      -- Performing Test HAS_FLTO - Success
      -- Found pybind11: /usr/local/include (found version "2.11.1")
      -- NOTE: rocPyDecode requires pybind11, Found --  /usr/local/include;/usr/include/python3.10 
      -- NOTE: rocPyDecode default installation path -- /usr/lib/python3.10 
      -- rocPyDecode -- Link Libraries: hip::device;/usr/lib/x86_64-linux-gnu/libavcodec.so;/usr/lib/x86_64-linux-gnu/libavformat.so;/usr/lib/x86_64-linux-gnu/libavutil.so;/opt/rocm/lib/librocdecode.so
      -- rocPyDecode -- CMAKE_CXX_FLAGS: -O3 -DNDEBUG -fPIC
      -- Copying /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/pyRocVideoDecode/__init__.py to rocPyDecode/pyRocVideoDecode/__init__.py
      -- Copying /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/pyRocVideoDecode/__init__.pyi to rocPyDecode/pyRocVideoDecode/__init__.pyi
      -- Copying /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/pyRocVideoDecode/decoder.py to rocPyDecode/pyRocVideoDecode/decoder.py
      -- Copying /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/pyRocVideoDecode/demuxer.py to rocPyDecode/pyRocVideoDecode/demuxer.py
      -- Copying /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/pyRocVideoDecode/types.py to rocPyDecode/pyRocVideoDecode/types.py
      -- ROCm Video Decode Library Python Binding - rocPyDecode module added 
      -- rocPyDecode Version -- 0.1.0
      -- rocPyDecode Install Path -- /usr/lib/python3.10
      /usr/bin/python3.10: No module named pip
      /usr/bin/python3.10: No module named pip
      -- Configuring done (0.5s)
      -- Generating done (0.0s)
      -- Build files have been written to: /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/build
      [ 11%] Building CXX object CMakeFiles/rocPyDecode.dir/src/roc_pybuffer.cpp.o
      [ 22%] Building CXX object CMakeFiles/rocPyDecode.dir/src/roc_pydecode.cpp.o
      [ 44%] Building CXX object CMakeFiles/rocPyDecode.dir/src/roc_pydlpack.cpp.o
      [ 44%] Building CXX object CMakeFiles/rocPyDecode.dir/src/roc_pyvideodecode.cpp.o
      [ 55%] Building CXX object CMakeFiles/rocPyDecode.dir/opt/rocm/share/rocdecode/utils/resize_kernels.cpp.o
      [ 66%] Building CXX object CMakeFiles/rocPyDecode.dir/opt/rocm/share/rocdecode/utils/colorspace_kernels.cpp.o
      [ 77%] Building CXX object CMakeFiles/rocPyDecode.dir/opt/rocm/share/rocdecode/utils/rocvideodecode/roc_video_dec.cpp.o
      [ 88%] Building CXX object CMakeFiles/rocPyDecode.dir/src/roc_pyvideodemuxer.cpp.o
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pybuffer.cpp:67:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
         67 | }
            | ^
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pyvideodecode.cpp:424:16: error: use of undeclared identifier 'CodecSupported'; did you mean 'PyCodecSupported'?
        424 |     bool ret = CodecSupported(device_id, codec_id, bit_depth);
            |                ^~~~~~~~~~~~~~
            |                PyCodecSupported
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pyvideodecode.cpp:423:31: note: 'PyCodecSupported' declared here
        423 | py::object PyRocVideoDecoder::PyCodecSupported(int device_id, rocDecVideoCodec codec_id, uint32_t bit_depth) {
            |                               ^
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pyvideodecode.cpp:424:10: error: no viable conversion from 'py::object' to 'bool'
        424 |     bool ret = CodecSupported(device_id, codec_id, bit_depth);
            |          ^     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      /usr/local/include/pybind11/detail/../detail/../pytypes.h:284:14: note: explicit conversion function is not a candidate
        284 |     explicit operator bool() const { return m_ptr != nullptr; }
            |              ^
      1 warning generated when compiling for gfx1030.
      2 errors generated when compiling for gfx1030.
      gmake[2]: *** [CMakeFiles/rocPyDecode.dir/build.make:118: CMakeFiles/rocPyDecode.dir/src/roc_pyvideodecode.cpp.o] Error 1
      gmake[2]: *** Waiting for unfinished jobs....
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pybuffer.cpp:67:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
         67 | }
            | ^
      1 warning generated when compiling for gfx1031.
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pybuffer.cpp:67:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
         67 | }
            | ^
      1 warning generated when compiling for gfx1032.
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pybuffer.cpp:67:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
         67 | }
            | ^
      1 warning generated when compiling for gfx1100.
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pybuffer.cpp:67:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
         67 | }
            | ^
      1 warning generated when compiling for gfx1101.
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pybuffer.cpp:67:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
         67 | }
            | ^
      1 warning generated when compiling for gfx1102.
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pybuffer.cpp:67:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
         67 | }
            | ^
      1 warning generated when compiling for gfx908.
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pybuffer.cpp:67:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
         67 | }
            | ^
      1 warning generated when compiling for gfx90a.
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pybuffer.cpp:67:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
         67 | }
            | ^
      1 warning generated when compiling for gfx940.
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pybuffer.cpp:67:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
         67 | }
            | ^
      1 warning generated when compiling for gfx941.
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pybuffer.cpp:67:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
         67 | }
            | ^
      1 warning generated when compiling for gfx942.
      /home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/src/roc_pybuffer.cpp:67:1: warning: non-void function does not return a value in all control paths [-Wreturn-type]
         67 | }
            | ^
      1 warning generated when compiling for host.
      gmake[1]: *** [CMakeFiles/Makefile2:128: CMakeFiles/rocPyDecode.dir/all] Error 2
      gmake: *** [Makefile:146: all] Error 2
      Traceback (most recent call last):
        File "/usr/local/lib/python3.10/dist-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/usr/local/lib/python3.10/dist-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
        File "/usr/local/lib/python3.10/dist-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
        File "/tmp/pip-build-env-03jh9pw9/overlay/local/lib/python3.10/dist-packages/setuptools/build_meta.py", line 332, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
        File "/tmp/pip-build-env-03jh9pw9/overlay/local/lib/python3.10/dist-packages/setuptools/build_meta.py", line 302, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-03jh9pw9/overlay/local/lib/python3.10/dist-packages/setuptools/build_meta.py", line 318, in run_setup
          exec(code, locals())
        File "<string>", line 73, in <module>
        File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
          raise CalledProcessError(retcode, cmd)
      subprocess.CalledProcessError: Command '['cmake', '--build', '/home/kiriti/develop/rocpydecode/rocpydecode-kiriti/rocpydecode/build', '--config', 'Release', '--parallel']' returned non-zero exit status 2.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.
kiriti@santiago:~/develop/rocpydecode/rocpydecode-kiriti/rocpydecode$ dpkg -l | grep rocm-core
ii  rocm-core                                  6.2.2.60202-116~22.04                             amd64        Radeon Open Compute (ROCm) Runtime software stack
```

### Operating System

Ubuntu 22.04

### CPU

ALL

### GPU

AMD Instinct MI300X

### ROCm Version

ROCm 6.2.0

**Comment by essamROCm** [comment]
@kiritigowda 
Hi Kiriti, thank you for looking at this.

The many warnings I see above were actually fixed by PR99, it was a function that is missing the return value.
But this error I see here about: (CodecSupported), is actually supposed not to be there if you use TOT rocDecode, because it was added recently to rocDecode. I saw such error before, but when I built/installed the rocDecode from TOT that error disappeared as the (CodecSupported) is present in the latest rocDecode code.

Can you please try to use the TOT rocDecode and try it again?

**Comment by rrawther** [comment]
@kiritigowda : Those functions are available only in tot/mainline build. develop rocPyDecode is not compatable with 6.2.2

**Comment by LakshmiKumar23** [comment]
@essamROCm we introduced this function https://github.com/ROCm/rocDecode/blob/0ba9992247a3beb096513435f8932a625da94d75/api/rocdecode_version.h#L50 that will go into rocm6.3, not sure if it made it to rocm6.2.2.

Until then we need to added a check like this I had added here https://github.com/ROCm/rocPyDecode/blob/5d386354b529efa784057f699db92ef7e092d186/src/roc_pyvideodecode.cpp#L433 to make sure the function is available in rocDecode version that comes with package. 

@kiritigowda  please confirm commit ID for rocDecode for rocm6.2.2, so then @essamROCm can use that method to check function availability

**Comment by kiritigowda** [comment]
@rrawther @LakshmiKumar23 -- as rocm 6.2.2 is the only latest public version available we should not fail to build TOT. Looks like there is a fix in PR #99 please confirm.

**Comment by kiritigowda** [comment]
> @kiritigowda Hi Kiriti, thank you for looking at this.
> 
> The many warnings I see above were actually fixed by PR99, it was a function that is missing the return value. But this error I see here about: (CodecSupported), is actually supposed not to be there if you use TOT rocDecode, because it was added recently to rocDecode. I saw such error before, but when I built/installed the rocDecode from TOT that error disappeared as the (CodecSupported) is present in the latest rocDecode code.
> 
> Can you please try to use the TOT rocDecode and try it again?

@essamROCm  the release version will always use the package dependencies. We can not have backward compatibility issues.

**Comment by LakshmiKumar23** [comment]
> @rrawther @LakshmiKumar23 -- as rocm 6.2.2 is the only latest public version available we should not fail to build TOT. Looks like there is a fix in PR #99 please confirm.

#99 only fixes the warnings. Not the `codecSupported` function call issue. The fix is to add a check

**Comment by essamROCm** [comment]
> @essamROCm we introduced this function https://github.com/ROCm/rocDecode/blob/0ba9992247a3beb096513435f8932a625da94d75/api/rocdecode_version.h#L50 that will go into rocm6.3, not sure if it made it to rocm6.2.2.
> 
> Until then we need to added a check like this I had added here
> 
> https://github.com/ROCm/rocPyDecode/blob/5d386354b529efa784057f699db92ef7e092d186/src/roc_pyvideodecode.cpp#L433
> 
> to make sure the function is available in rocDecode version that comes with package.
> @kiritigowda please confirm commit ID for rocDecode for rocm6.2.2, so then @essamROCm can use that method to check function availability

@LakshmiKumar23 @kiritigowda @rrawther 
Lakshmi, are you sure of this, because the #if check is actually surrounding different code below, not the actual "CodecSupported" function, which is above that #if check, can you please confirm?

I mean should I use "#if ROCDECODE_CHECK_VERSION(0,6,0)" or "#if OVERHEAD_SUPPORT"?
or:  "#if ROCDECODE_CHECK_VERSION(0,6,3)"
Also, this is actually echos in the API as well, and the documentation, should I just remove the whole codec supported thing? which if I do the original error they reported will come back? So not sure what to do !!

**Comment by LakshmiKumar23** [comment]
> > @essamROCm we introduced this function https://github.com/ROCm/rocDecode/blob/0ba9992247a3beb096513435f8932a625da94d75/api/rocdecode_version.h#L50 that will go into rocm6.3, not sure if it made it to rocm6.2.2.
> > Until then we need to added a check like this I had added here
> > https://github.com/ROCm/rocPyDecode/blob/5d386354b529efa784057f699db92ef7e092d186/src/roc_pyvideodecode.cpp#L433
> > 
> > to make sure the function is available in rocDecode version that comes with package.
> > @kiritigowda please confirm commit ID for rocDecode for rocm6.2.2, so then @essamROCm can use that method to check function availability
> 
> @LakshmiKumar23 Lakshmi, are you sure of this, because the #if check is actually surrounding different code below, not the actual "CodecSupported" function, which is above that #if check, can you please confirm?

@essamROCm  I mean a similar  #if check is needed for `codecSupported` also, right now I think it is missing

**Comment by essamROCm** [comment]
Even if I 'check' in C++ side, it is going be difficult at the python API side. So I am going to keep the API, but it does nothing if the #if check made in the C++ code side.

Now, how to check if the version is 6.2.2 not above?

**Comment by LakshmiKumar23** [comment]
> Even if I 'check' in C++ side, it is going be difficult at the python API side. How to check if the version is 6.2.2 not above?

@essamROCm please refer this
https://github.com/ROCm/rocPyDecode/blob/6a315dc3107d41f9e8e8626badc0bc92137317a1/pyRocVideoDecode/decoder.py#L145

for c++ it's (major,minor, patch) as in the function so (6,3,0). It will work only after that function is pushed to mainline so for rocm6.2.2 public, this won't work

**Comment by essamROCm** [comment]
@kiritigowda 
Fixed in PR: https://github.com/ROCm/rocPyDecode/pull/105

**Comment by essamROCm** [comment]
Fixed in PR#105
